### PR TITLE
Added ability to use many NS servers

### PR DESCRIPTION
**Pull request by yuri-zubov**
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [x] Update Changelog following the conventions laid out [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [ ] RuboCop passes

- [ ] Existing tests pass

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

### CODE REVIEW

File: bin/check-dns.rb [hunk 1]
```diff
@@ -54,7 +54,7 @@ class DNS < Sensu::Plugin::Check::CLI
          default: 'IN'
 
   option :server,
-         description: 'Server to use for resolution',
+         description: 'A list of servers to use for resolution (comma separated list)',
```
Comment (mbbroberg) [inline bin/check-dns.rb]: Minor edit: `A comma-separated list of servers to use for resolution`

**Comment by mbbroberg** [comment]
Rubocop is failing as well ([errors are here](https://travis-ci.org/sensu-plugins/sensu-plugins-dns/jobs/348637727)). We'll want to get that passing too. 

[@majormoses did a thoughtful write up](https://blog.sensuapp.org/contributing-to-sensu-updating-rubocop-to-resolve-cve-2017-8418-7b9a7bb8d9c) on how to configure your environment to review rubocop errors if you want to use that as a guide. Note that you don't have to handle the rubocop upgrade to `0.51` as part of this, but you're welcome to if it helps.

Let us know if you have any questions! I'd love to see your improvement merge 👌

**Comment by yuri-zubov** [comment]
from my perspective, it's should be an option list (not a fallback). If provided, we want to check against the resolvers. If not, we use the system resolvers

**Comment by yuri-zubov** [comment]
test
```
bin/check-dns.rb -d actility.com -s dasdasddasd.com,8.8.8.8
DNS CRITICAL: 50% of tests succeeded: Could not resolve actility.com: Query timed out
```
```
bin/check-dns.rb -d actility.com
DNS OK: Resolved actility.com A
```
```
bin/check-dns.rb -d actility.com -s 8.8.4.4,8.8.8.8        
DNS OK: Resolved actility.com A
```

File: bin/check-dns.rb [hunk 2]
```diff
@@ -184,7 +185,7 @@ def check_results(entries)
         # consistency reasons
         if config[:type] == 'A' || config[:type] == 'AAAA'
           check_ips(entries)
-        # non ip type
+          # non ip type
```
Comment (majormoses) [inline bin/check-dns.rb]: this should not be indented it is explaining that the else is for non ip type records.

**Comment by majormoses** [comment]
> from my perspective, it's should be an option list (not a fallback). If provided, we want to check against the resolvers. If not, we use the system resolvers

That's fine to check all of them they are just different use cases and if someone needs that then they can add it in the future was just checking if you had a use case for it.

> If not, we use the system resolvers

Can you explain this? I see nowhere in the code that falls back to the local resolvers.

**Comment by yuri-zubov** [comment]
For the change,  yes go with it
For the 1st question, we have a use case as it happened that some Authoritative DNS were not properly updated, generating some issues on other services

For the 2nd question,
look code(if option isn't sent nameserver will not set)
```
 ns = config[:server].nil? ? [nil] : config[:server].split(',')
dnsruby_config[:nameserver] = server unless server.nil?
```
and dnsruby will take default NS(system NS)

**Comment by majormoses** [comment]
I must have missed that.

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-dns/versions/1.4.0

**Pull request merged**